src/main/scala/beam/agentsim/agents/modalbehaviors/ChoosesMode.scala

```diff
@@ -1463,10 +1522,10 @@ trait ChoosesMode {
               vehiclesUsed.headOption.filter(mustBeDrivenHome).map(_.id)
             else
               data.personData.currentTourPersonalVehicle
-                .orElse(vehiclesUsed.headOption.filter(mustBeDrivenHome).map(_.id))
+                .orElse(vehiclesUsed.headOption.filter(mustBeDrivenHome).map(_.id)),
+          failedTrips = data.personData.failedTrips ++ data.personData.currentTrip
```

Comment:
If I am reading this correct, then any trip that is not a teleportation event should be considered a failure?

Reply:
Generally, this is true. If we have some trip defined (`currentTrip` is an `Option`) that we cannot finish (we're choosing a new trip now (in `FinishingModeChoice` state)).
About teleportation, I'm not sure but it seems that they suppose that no teleportation failure is possible.

Reply:
I will defer to @zneedell on that aspect - and he already approved - so otherwise LGTM